plugin-development/api-plugin-input.md

```diff
@@ -102,7 +102,31 @@ router.emit(tag, time, {:foo => 'bar'})
 
 ## Methods
 
-There are no specific methods for the Input plugins.
+### zero_downtime_restart_ready?
+
+To support [Zero-downtime restart](../deployment/zero-downtime-restart.md), you can override this method to return `true`.
+
+```ruby
+def zero_downtime_restart_ready?
+  true
+end
+```
+
+To do this, the following condition must be met:
+
+* This plugin can run in parallel with another Fluentd.
+
+This is because there is a period when the old process and the new process run in parallel during a zero-downtime restart.
+
+After addressing the following considerations and ensuring there are no issues, override this method.
+Then, the plugin will not experience downtime with zero-downtime restart.
```

Comment:
```suggestion
Then, the plugin will succeed with zero-downtime restart.
```

It could be simple enough?

Reply:
Thanks!
I have a little concern about whether the “succeed” is appropriate here, but I think it makes sense, so let's go with that!